Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManagerTest.java

```diff
@@ -0,0 +1,89 @@
+package edu.asu.spring.quadriga.service.impl.workbench;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import edu.asu.spring.quadriga.dao.workbench.IRetrieveProjectDAO;
+import edu.asu.spring.quadriga.domain.enums.EProjectAccessibility;
+import edu.asu.spring.quadriga.domain.proxy.ProjectProxy;
+import edu.asu.spring.quadriga.domain.workbench.IProject;
+import edu.asu.spring.quadriga.dto.ProjectDTO;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.workbench.mapper.IProjectShallowMapper;
+
+public class RetrieveProjectManagerTest {
+
+	@InjectMocks
+	private RetrieveProjectManager retrieveProjectManagerUnderTest;
+
+	@Mock
+	private IRetrieveProjectDAO mockedDBConnect = Mockito.mock(IRetrieveProjectDAO.class);
+
+	@Mock
+	private IProjectShallowMapper mockedProjectShallowMapper = Mockito.mock(IProjectShallowMapper.class);
+
+	@Before
+	public void setUp() throws QuadrigaStorageException {
+		MockitoAnnotations.initMocks(this);
+	}
+
+	@Test
+	public void checkOnlyPublicProjectsAreReturned() throws QuadrigaStorageException{
+		Mockito.when(mockedDBConnect.getAllProjectsDTOByAccessibility("PUBLIC")).thenReturn(createPublicProjectDTOList());
+		Mockito.when(mockedProjectShallowMapper.getProjectDetails(Matchers.any(ProjectDTO.class)))
+		.thenReturn(createPublicIProject());
+		List<IProject> projectList = retrieveProjectManagerUnderTest.getProjectListByAccessibility("PUBLIC");
+
+		Mockito.verify(mockedDBConnect).getAllProjectsDTOByAccessibility("PUBLIC");
+		Mockito.verify(mockedProjectShallowMapper).getProjectDetails(Matchers.any(ProjectDTO.class));
+		Mockito.verify(mockedDBConnect, times(1)).getAllProjectsDTOByAccessibility("PUBLIC");
```

Comment:
you only need this line or line 48. You basically test for the same thing twice.